### PR TITLE
Adds dynamic disk types

### PR DESCRIPTION
**Pull request by ashraffouda**
Signed-off-by: Ashraf Fouda <ashraf.m.fouda@gmail.com>

### CODE REVIEW

File: apps/cbportal/base/CBGrid/.macros/page/add_disktype/3_add_disktype.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from JumpScale.portal.docgenerator.popup import Popup
```
Comment (grimpy) [inline apps/cbportal/base/CBGrid/.macros/page/add_disktype/3_add_disktype.py]: why is this a macro and not defined in actions?
I dont see anything dynamic here.

File: apps/cbportal/base/CBGrid/Disk Types/Disk Types.wiki [hunk 2]
```diff
@@ -0,0 +1,8 @@
+
+h2. $$page
+
+#{{pageexplorer}}
+
+h2. children
+
+{{children: bullets}}
```
Comment (grimpy) [inline apps/cbportal/base/CBGrid/Disk Types/Disk Types.wiki]: Can we delete this again?

File: apps/cbportal/base/CBGrid/.macros/page/add_disktype/3_add_disktype.py (hunk 1 above)
Comment (ashraffouda) [reply apps/cbportal/base/CBGrid/.macros/page/add_disktype/3_add_disktype.py]: https://github.com/0-complexity/openvcloud/pull/1805/files#diff-f170d4d45e39377368fb3a3

Comment (ashraffouda) [reply apps/cbportal/base/CBGrid/.macros/page/add_disktype/3_add_disktype.py]: https://github.com/0-complexity/openvcloud/pull/1805/files#diff-f170d4d45e39377368fb3a398719ef78R10

File: apps/cloudbroker/base/cloudapi__disks/methodclass/cloudapi_disks.py [hunk 3]
```diff
@@ -1,3 +1,4 @@
+from __future__ import division
```
Comment (grimpy) [inline apps/cloudbroker/base/cloudapi__disks/methodclass/cloudapi_disks.py]: why is this import here?

File: apps/cloudbroker/base/cloudapi__disks/methodclass/cloudapi_disks.py [hunk 4]
```diff
@@ -382,3 +394,84 @@ def deleteDisks(self, diskIds, reason, permanently=False, **kwargs):
...
+    def addType(self, id, description, vpool=None, cacheratio=None, snapshotable=True, **kwargs):
+        if cacheratio <= 0 and vpool:
+            raise exceptions.BadRequest(
+                "Cache Ratio must be larger than Zero"
+            )
+        dtype = self.models.disktype.searchOne({'id': id})
+        if dtype:
+            raise exceptions.BadRequest(
+                "Disk type with id: {} already exits, Please choose other name".format(id)
+            )
```
Comment (grimpy) [inline apps/cloudbroker/base/cloudapi__disks/methodclass/cloudapi_disks.py]: you need to make a check there is not type with same vpool (which is assume is the prefix?)

File: apps/cloudbroker/base/cloudapi__disks/methodclass/cloudapi_disks.py [hunk 5]
```diff
@@ -382,3 +394,84 @@ def deleteDisks(self, diskIds, reason, permanently=False, **kwargs):
...
+        dtype = self.models.disktype.searchOne({'id': id})
+        if not dtype:
+            raise exceptions.BadRequest(
+                "Can not find Disk Type with id: {}".format(id)
+            )
+        
+        if cacheratio <= 0 and (dtype['vpool'] or vpool):
+            raise exceptions.BadRequest(
+                "Cache Ratio must be larger than Zero"
+            )
```
Comment (grimpy) [inline apps/cloudbroker/base/cloudapi__disks/methodclass/cloudapi_disks.py]: here you need to check for vpool aswell

File: libs/CloudscalerLibcloud/CloudscalerLibcloud/compute/drivers/libvirt_driver.py [hunk 6]
```diff
@@ -339,7 +349,7 @@ def getBestDataVpool(self):
             )
         if len(diskspervpool) > 1:
             for vpool in list(diskspervpool.keys()):
-                if not vpool.startswith("data"):
+                if not vpool.startswith(prefix):
```
Comment (grimpy) [inline libs/CloudscalerLibcloud/CloudscalerLibcloud/compute/drivers/libvirt_driver.py]: Geert says it needs to match the pattern prefix\d{1,2} (prefix + 2 digits)
